data/api/client-server/profile.yaml

```
@@ -67,163 +104,140 @@ paths:
       tags:
         - User data
     get:
-      summary: Get the user's display name.
+      summary: Get the user's profile field.
       description: |-
-        Get the user's display name. This API may be used to fetch the user's
-        own displayname or to query the name of other users; either locally or
-        on remote homeservers.
-      operationId: getDisplayName
+        Get the value of a user's profile field. This API can retrieve the profile fields of the user
+        or other users, either locally or on remote homeservers. It supports both standard fields
+        (e.g. "displayname" and "avatar_url") and custom keys.
+      operationId: getProfileField
       parameters:
         - in: path
           name: userId
-          description: The user whose display name to get.
+          description: The user whose profile field to get.
           required: true
           example: "@alice:example.com"
           schema:
             type: string
+        - in: path
+          name: key_name
+          description: The profile field key name to retrieve (e.g. "displayname", "avatar_url", or custom).
```

```suggestion
          description: The profile field key name to retrieve
```

I sort of feel like the `e.g.` is overkill as we list it below?

---

Good point! I snuck it into this, thanks: https://github.com/matrix-org/matrix-spec/pull/2071/commits/ee9b5ddcca0264866e6179c1d761c1a518098b52